mlx/primitives.h

```diff
@@ -856,6 +856,30 @@ class Floor : public UnaryPrimitive {
   void eval(const std::vector<array>& inputs, array& out);
 };
 
+class FastInferenceSDPA : public UnaryPrimitive {
```

Comment:
Nit: Based on the name we go for in the high-level op API, I think it makes sense to also change the name of this primitive to reflect that.

Reply:
yes, we're going to use 

scaled_dot_product_attention;  and 
ScaledDotProductAttention

and put in the mx.fast namespace (python, & c analogue etc., fleshing out the details locally)